### DISCUSSION TITLE
Ver 0.141 Heatmap and Ambient Mode not working on keyboard (HQ403TS-G513QM)

### DISCUSSION THREAD

**Discussion by RAGE-HERO7** [seerge/g-helper#1781] [General]
*Describe the problem
Ambient and Heatmap light modes are not working on Keyboard (The light only turns on in lightbar but won't turn on in keyboard ) it was working perfectly in GHELPER ver 0.134

![20231224_133729](https://github.com/seerge/g-helper/assets/154586512/e289b10b-6031-4adc-a92a-714e773f1dcb)
![20231224_134206](https://github.com/seerge/g-helper/assets/154586512/1bf3de1a-d8eb-4618-a9cc-96456b69e26a)

*How ro recreate the problem
Try to set the Heatmap or ambient light mode via Keyboard shortcut or manually inside GHELPER

LOG FILE
[log.txt](https://github.com/seerge/g-helper/files/13761360/log.txt)

*Additional note on problems that got solved 
   with Ver 0.141
GPU light mode won't turn on when selected and Keyboard and lightbar lights won't turn off in sleep mode is solved Thank you for fixing that @seerge 🙏

Device - Asus Rog Strix G15 2021
Model No - (HQ403TS-G513QM)

Op System - Windows 11 22H2 Enterprise Ed

Spec - Amd Ryzen 9 5900HX, Nvidia Rtx 3060, 16gb ddr4 kingston Ram, 1tb Intel 670p ssd, 2k@165HZ

*GHELPER VER 0.141
*No Armoury crate,my asus app installed,
*All asus service's are stopped
*All device drivers and Bios (ver.331) are uptodate

**Comment by seerge** [comment]
@RAGE-HERO7 you  told me that your device doesn't have advanced RGB options before, so I have added your model to a list of a ones with limited RGB support. Where lightning is applied per zone.

Does Ambient work in 0.137 ? 

I have removed (again) your model from the list in this build
[GHelper.zip](https://github.com/seerge/g-helper/files/13761489/GHelper.zip)

**Reply by RAGE-HERO7** [reply]
@seerge

* My device does have a working RGB Lightbar and keyboard that support per key RGB just like any other Asus Rog G15 released on 2021

* what my device doesn't support is RGB control on Logo that located on the surface of LID and my device doesn't have a Lightbar that goes on the bottom backside of the LID like in the higher rog strix scar variants.

*No sir, i didn't tried ver 0.137
  Tried 0.134 and updated straight to 0.140
   But while i was using ver 0.134 all the light modes including (Ambient,heatmap,gpu mode) were working perfectly without any problem


* I Sincerely apologize for giving wrong information / for not clearly mentioning the issues on my previous posts

**Reply by seerge** [reply]
Does Ambient in the build I have linked here work for you?

**Reply by RAGE-HERO7** [reply]
Sorry i am currently not present at my home rt now :(    but tomorrow i will download and test the version you linked.....Thanku :)

**Reply by RAGE-HERO7** [reply]
@seerge

*i tried the ver you linked above it is working fine now...everything is back to normal Thank you so much🫶

*Also i UNDERVOLTTED my
CPU Ryzen 9 5900hx to -20 on all 3 modes and it is working. I didn't saw this CPU in your undervolt supported cpu list section 😅!!


https://github.com/seerge/g-helper/assets/154586512/cea6d272-59ca-49e6-9f40-c19ce67d1440
![20231224_224001](https://github.com/seerge/g-helper/assets/154586512/9f7d7b0d-4ed9-4759-a1d2-ca7e4a96727e)
![20231224_224049](https://github.com/seerge/g-helper/assets/154586512/0b671353-3096-40d9-a258-35b8ad314d6d)

LOG FILE
[log.txt](https://github.com/seerge/g-helper/files/13762475/log.txt)

**Comment by pberenyi** [comment]
I have the same issue on a Zephyrus G15 GA503RM. The heatmap mode was working on the previous stable release, but stopped working after the upgrade to 0.141. v0.143 seems to fix the issue. Heatmap is working on that again.

**Comment by Sawa-i** [comment]
I'mma up this here with ASUS ROG Strix G713PV (2023), 

Heatmap and Ambient causes  side lightbars to light up.

But Keyboard stays off.

**Reply by seerge** [reply]
@Sawa-i does your device have per-key RGB or 4-zone RGB?

**Reply by Sawa-i** [reply]
Per key.

And again mentioning the light bars at front-side below (similar to thread poster's)

On Ambient and Heatmap, only light bars light up... And they seem to be alright according to what's on screen or Cpu temps...

Btw, GPU Temperature is fine

**Reply by RAGE-HERO7** [reply]
for me how @seerge permanently solved this problem is by this 👇

![Screenshot_20240426_001056_WhatsApp](https://github.com/seerge/g-helper/assets/154586512/704e3ef4-b2c7-411f-84ef-fe1a92dbdd4f)

ever since i added "per_key_rgb" : 1, to jason.config i never encountered problem with lighting modes

**Reply by Sawa-i** [reply]
Daym man. You just resolved it.

@RAGE-HERO7  & @seerge  thanks for your assistance.

Btw Important thing I'd like to tell you guys, adding this "Key-value" pair in config.json also unlocked rest of my Laptop Keyboard Lighting Options (that  were already available in Armoury Crate, e.g. Laser, Star, Ripple, etc.)

Also for people looking for 'config.json' , it's in Your Hidden "AppData" folder (where within it ./Roaming/GHelper was where it was).

Overalll Path: Local Disk C (Windows Partition) / Users / $Username / AppData (Hidden Folder) / Roaming / GHelper

**Reply by seerge** [reply]
@Sawa-i correct, your model G713PV is listed as one that has limited 4-zone RGB, and therefore app would try to set that for it by default.

Apparently your device same as @RAGE-HERO7 's exist in 2 different editions for SAME model name.

So setting it manually - is the solution :)
